components/definition/stimulation/NetPyNEStimulationSource.js

```diff
@@ -10,79 +10,259 @@ import RaisedButton from 'material-ui/RaisedButton';
 import clone from 'lodash.clone';
 import Utils from '../../../Utils';
 import NetPyNEField from '../../general/NetPyNEField';
+import ListComponent from '../../general/List';
 
 var PythonControlledCapability = require('../../../../../js/communication/geppettoJupyter/PythonControlledCapability');
 var PythonControlledTextField = PythonControlledCapability.createPythonControlledControl(TextField);
+var PythonControlledSelectField = PythonControlledCapability.createPythonControlledControl(SelectField);
+var PythonControlledListComponent = PythonControlledCapability.createPythonControlledControl(ListComponent);
 
 export default class NetPyNEStimulationSource extends React.Component {
 
   constructor(props) {
     super(props);
-    var _this = this;
     this.state = {
-      currentName: props.name
+      currentName: props.name,
+      sourceType: null
     };
+    
+    this.stimSourceTypeOptions = [
+      { type: 'IClamp' }, 
+      { type: 'VClamp' }, 
+      { type: 'SEClamp'},
+      { type: 'NetStim'},
+      { type: 'AlphaSynapse'}
+    ];
+    this.handleStimSourceTypeChange = this.handleStimSourceTypeChange.bind(this);
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (this.state.currentName!=nextProps.name) {
+      this.setState({ currentName: nextProps.name, sourceType: null});
+    };
+  };
+  
   handleRenameChange = (event) => {
     var that = this;
     var storedValue = this.props.name;
     var newValue = event.target.value;
     this.setState({ currentName: newValue });
     this.triggerUpdate(function () {
       // Rename the population in Python
-      Utils.renameKey('netParams.stimSourceParams', storedValue, newValue, (response, newValue) => { that.renaming=false;});
-      that.renaming=true;
+      Utils.renameKey('netParams.stimSourceParams', storedValue, newValue, (response, newValue) => { that.renaming = false });
+      that.renaming = true;
```

Comment:
Do we actually need 'that'? What's the value of 'this' inside that method?

Reply:
I saw in NetPyNePopulation.js that code to handle population rename, so I used the same one. Maybe we could do it differently.